src/idpi/metadata.py

```diff
@@ -11,12 +11,38 @@
 import xarray as xr
 from earthkit.data.writers import write  # type: ignore
 
+# Local
+from . import grib_decoder
+
+VCOORD_TYPE = {
+    "generalVertical": ("model_level", -0.5),
+    "generalVerticalLayer": ("model_level", 0.0),
+    "isobaricInPa": ("pressure", 0.0),
+}
+
+
+def extract(metadata):
+    [vref_flag] = grib_decoder.get_code_flag(
+        metadata.get("resolutionAndComponentFlags"), [5]
+    )
+    level_type = metadata.get("typeOfLevel")
+    vcoord_type, zshift = VCOORD_TYPE.get(level_type, (level_type, 0.0))
+
+    return {
+        "parameter": metadata.as_namespace("parameter"),
+        "geography": metadata.as_namespace("geography"),
+        "vref": "native" if vref_flag else "geo",
+        "vcoord_type": vcoord_type,
+        "origin_z": zshift,
+    }
+
 
 def override(message: bytes, **kwargs: typing.Any) -> dict[str, typing.Any]:
     """Override GRIB metadata contained in message.
 
     Note that no special consideration is made for maintaining consistency when
     overriding template definition keys such as productDefinitionTemplateNumber.
+    Note that the origin components in x and y will be unset.
```

Comment:
I cant see where in `override` or `extract` origin_x or origin_y are unset. Does this happen in earthkit-data's GribMetadata override method?

Reply:
Yeah this note is a bit outdated, in the current implementation, the origin_xy values just remain untouched. I'll update the note

Reply:
Yes thanks, since untouched and unset have different meanings it would be good to update